extensions-core/hdfs-storage/src/main/java/org/apache/hadoop/fs/HadoopFsWrapper.java

```diff
@@ -0,0 +1,61 @@
+/*
+* Licensed to Metamarkets Group Inc. (Metamarkets) under one
+* or more contributor license agreements. See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership. Metamarkets licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.apache.hadoop.fs;
+
+import io.druid.java.util.common.logger.Logger;
+
+import java.io.IOException;
+
+/**
+ * This wrapper class is created to be able to access the "protected" method inside Hadoop's FileSystem class.
+ * Same is supposed to be made public in hadoop eventually.
+ */
+public class HadoopFsWrapper
+{
+  private static final Logger log = new Logger(HadoopFsWrapper.class);
+
+  private FileSystem fs;
+
+  public HadoopFsWrapper(FileSystem fs) {
+    this.fs = fs;
+  }
+
+  /**
+   * Same as FileSystem.rename(from, to, Options.Rename.NONE) . That is,
+   * it returns "false" when "to" directory already exists. It is different from FileSystem.rename(from, to)
+   * which moves "from" directory inside "to" directory if it already exists.
+   *
+   * @param from
+   * @param to
+   * @return
+   * @throws IOException
+   */
+  public boolean rename(Path from, Path to) throws IOException
```

Comment:
my intention is not to create a generic FileSystem wrapper really unless necessary... we can do that when there is a need.